### PR TITLE
Adjustments for app-sre deployment

### PR DESCRIPTION
**Pull request by bsquizz**

### CODE REVIEW

File: backend/ibutsu_server/__init__.py [hunk 1]
```diff
@@ -35,6 +35,13 @@ def _make_sql_url(hostname, database, **kwargs):
     return "postgresql://{}/{}".format(url, database)
 
 
+def _make_broker_url(env_var_value, hostname, password, port):
+    if env_var_value:
+        return env_var_value
+    user_pass_str = ":{}@".format(password) if password else ""
+    return "redis://{}{}:{}".format(user_pass_str, hostname, port)
```
Comment (john-dupuy) [inline backend/ibutsu_server/__init__.py:42]: Can we change this to use `f`-strings?

File: ocp-templates/app-interface/flower.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          deploymentConfig: celery-flower
+      spec:
+        containers:
+        - env:
+          # TOOD: use redis secret to set this
+          - name: BROKER_URL
+            value: redis://:${REDIS_PASSWORD}@redis.${NAMESPACE}.svc
+          image: celery-flower
+          imagePullPolicy: IfNotPresent
+          name: quay.io/ibutsu/celery-flower:${IMAGE_TAG}
```
Comment (john-dupuy) [inline ocp-templates/app-interface/flower.yaml:35]:
```suggestion
          image: quay.io/ibutsu/flower:${IMAGE_TAG}
          imagePullPolicy: Always
```

Comment (bsquizz) [reply ocp-templates/app-interface/flower.yaml]: Adjusted image name but the pull policy should stay at 'IfNotPresent' -- it's a recommended best practice for AppSRE

File: ocp-templates/app-interface/backend.yaml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    strategy:
+      type: Rolling
+    template:
+      metadata:
+        labels:
+          app: ${APP_NAME}
+          deploymentConfig: ibutsu-backend
+      spec:
+        containers:
+        - env:
```
Comment (john-dupuy) [inline ocp-templates/app-interface/backend.yaml:29]: I'm happy to do this as a follow-up, but there is one more secret env var we need to run Ibutsu + Auth:
https://github.com/ibutsu/ibutsu-server/issues/252

File: backend/ibutsu_server/__init__.py [hunk 4]
```diff
@@ -64,7 +71,19 @@ def get_app(**extra_config):
...
+                    config.get("REDIS_HOSTNAME"),
+                    config.get("REDIS_PASSWORD"),
+                    config.get("REDIS_PORT"),
+                ),
+                "CELERY_RESULT_BACKEND": _make_broker_url(
+                    config.get("CELERY_RESULT_BACKEND"),
+                    config.get("REDIS_HOSTNAME"),
+                    config.get("REDIS_PASSWORD"),
+                    config.get("REDIS_PORT"),
+                ),
```
Comment (john-dupuy) [inline backend/ibutsu_server/__init__.py:86]: Can we move these into their own `if` block? 

```python
if config.get("CELERY_BROKER_URL") or config.get("REDIS_HOSTNAME"):
```

```python
if config.get("CELERY_RESULT_BACKEND") or config.get("REDIS_HOSTNAME"):
```

Comment (bsquizz) [reply backend/ibutsu_server/__init__.py:86]: Ah, didn't catch that. Yep.

**Review by john-dupuy — APPROVED** [review]

**Comment by bsquizz** [comment]
@john-dupuy @rsnyman you guys ok w/ merging this? I think the main impact would be just making sure that the value of the CELERY_BROKER_URL still gets read in and used for redis as expected on the existing PSI deployment

**Comment by john-dupuy** [comment]
Yup, I think we're good to merge this in.

**Pull request merged**